rundeckapp/grails-app/conf/RefererFilters.groovy

```diff
@@ -0,0 +1,31 @@
+class RefererFilters {
+    def grailsApplication
+
+    def filters = {
+        checkReferer(controller: '*', action: '*') {
+            before = {
+                def csrf = grailsApplication.config.rundeck.security.csrf
+                if(csrf && csrf != 'NONE'){
+                    if(csrf == 'POST') {
+                        if (request.method.toUpperCase() == "POST") {
+                            // referer must match serverURL, optionally https
+                            def validRefererPrefix = "^${grailsApplication.config.grails.serverURL}".replace("http", "https?")
+                            def referer = request.getHeader('Referer')
+                            return referer && referer =~ validRefererPrefix
+                        }
+                    }else if(csrf == '*'){
+                        def validRefererPrefix = "^${grailsApplication.config.grails.serverURL}".replace("http", "https?")
+                        def referer = request.getHeader('Referer')
+                        if(request.method.toUpperCase() == 'GET' && (referer == null || referer.endsWith("user/login"))) {
```

Comment:
instead of checking referer.endsWith, should check the `controllerName` and `actionName` variables.  Or, even better, change line 5 to exclude that pattern